src/app/stake/page.tsx

```diff
@@ -74,8 +75,13 @@ export default function Stake() {
     "symbols"
   );
 
+  const stakingSupportedChainIds = addressesChainIds.filter((chainId) => {
+    const chain = chainsDetails?.find((c) => c.id === chainId);
+    return chain && isStakingSupported(chain);
+  });
```

Comment:
A bit simpler + [useMemo](https://react.dev/reference/react/useMemo):
```suggestion
  const stakingSupportedChainIds = useMemo(() => chainsDetails?.find((chain) => isStakingSupported(chain)).map((chain) => chain.id);
  }),
  [chainDetails]);
```